azurerm/internal/services/network/express_route_port_resource.go

```diff
@@ -0,0 +1,514 @@
+				Computed: true,
+			},
+			"interface_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"patch_panel_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"rack_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"connector_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	},
+}
+
+func resourceArmExpressRoutePort() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmExpressRoutePortCreateUpdate,
+		Read:   resourceArmExpressRoutePortRead,
+		Update: resourceArmExpressRoutePortCreateUpdate,
+		Delete: resourceArmExpressRoutePortDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ExpressRoutePortID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ExpressRoutePortName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": location.Schema(),
+
+			"peering_location": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"bandwidth_in_gbps": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+
+			"encapsulation": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.Dot1Q),
+					string(network.QinQ),
+				}, false),
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"identity_ids": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MinItems: 1,
+							Elem: &schema.Schema{
+								Type:             schema.TypeString,
+								ValidateFunc:     validation.NoZeroValues,
+								DiffSuppressFunc: suppress.CaseDifference,
+							},
+						},
+
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							// TODO: The "ignoreCase" and diff suppression function can be removed once
+							// following issue get resolved:
+							// https://github.com/Azure/azure-rest-api-specs/issues/12330
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.ResourceIdentityTypeUserAssigned),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+					},
+				},
+			},
+
+			"link1": expressRoutePortSchema,
+
+			"link2": expressRoutePortSchema,
+
+			"ether_type": {
```

Comment:
is this short for ethernet_type?
```suggestion
			"ethernet_type": {
```

Reply:
EtherType is a term it self, see: https://en.wikipedia.org/wiki/EtherType and https://docs.microsoft.com/en-us/azure/expressroute/expressroute-erdirect-about#technical-requirements, so we shall keep it as this.

Reply:
huh well TIL! https://en.wikipedia.org/wiki/EtherType

given the wiki and there being no space between the words i think it should be ethertype?
```suggestion
			"ethertype": {
```

Reply:
normally yes, but i think in this instance it makes more sense to not add the _ as its not two words but one